### PR TITLE
feat: add website

### PR DESCRIPTION
**Pull request by sandren**
Renew the website based on #152.

### CODE REVIEW

File: packages/website/.npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+strict-peer-dependencies=false
```
Comment (dai-shi) [inline packages/website/.npmrc:1]: Is this file necessary in the repo? If we can remove it, let's remove it for now.

File: packages/website/.nvmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+18
```
Comment (dai-shi) [inline packages/website/.nvmrc:1]: Please remove this file.

File: packages/website/package.json [hunk 3]
```diff
@@ -1,6 +1,11 @@
 {
   "name": "waku-website",
-  "version": "0.1.0",
+  "version": "0.0.0",
```
Comment (dai-shi) [inline packages/website/package.json:3]: We don't need to change. It's private anyway though.
```suggestion
  "version": "0.1.0",
```

File: packages/website/package.json [hunk 4]
```diff
@@ -1,6 +1,11 @@
...
+  "description": "⛩️ The minimal React framework",
+  "homepage": "https://github.com/dai-shi/waku",
+  "bugs": "https://github.com/dai-shi/waku/issues",
+  "license": "MIT",
```
Comment (dai-shi) [inline packages/website/package.json:7]: Not required here.

File: packages/website/package.json [hunk 5]
```diff
@@ -9,14 +14,19 @@
...
-    "autoprefixer": "^10.4.16",
-    "bright": "^0.8.4",
     "express": "^4.18.2",
-    "glob": "^10.3.10",
     "react": "18.3.0-canary-a41957507-20231017",
     "react-dom": "18.3.0-canary-a41957507-20231017",
     "react-server-dom-webpack": "18.3.0-canary-a41957507-20231017",
-    "tailwindcss": "^3.3.3",
-    "waku": "workspace:*"
+    "waku": "0.16.0"
```
Comment (dai-shi) [inline packages/website/package.json:21]:
```suggestion
    "waku": "workspace:*"
```
We use the workspace protocol in monorepo.

File: packages/website/prettier.config.cjs [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  jsxSingleQuote: false,
+  printWidth: 100,
+  quoteProps: 'consistent',
+  semi: true,
+  singleQuote: true,
+  tabWidth: 2,
+  trailingComma: 'all',
+  tailwindConfig: './tailwind.config.js',
+  tailwindFunctions: ['cx'],
+};
```
Comment (dai-shi) [inline packages/website/prettier.config.cjs:16]: I think the prettier config should be the same across the monorepo. It may require to install prettier-plugin-tailwindcss in root, but that can't be helped, I guess.

File: packages/website/vite.config.ts [hunk 7]
```diff
@@ -1,23 +1,23 @@
...
+        output: {
+          preserveModules: true,
+          preserveModulesRoot: modulesRoot,
+        },
       },
     },
-  },
-});
+  });
+};
```
Comment (dai-shi) [inline packages/website/vite.config.ts:23]: We can probably remove this file, because now the website doesn't use waku/router.

**Review by dai-shi — CHANGES_REQUESTED** [review]
We need slight changes for setup/env/etc.

File: packages/website/tailwind.config.js [hunk 8]
```diff
@@ -1,14 +1,39 @@
+const colors = require("tailwindcss/colors");
+
 /** @type {import('tailwindcss').Config} */
-export default {
-  content: ["./index.html", "./{src,routes}/**/*.{ts,tsx}"],
+module.exports = {
```
Comment (dai-shi) [inline packages/website/tailwind.config.js:4]: Does this work? Everything should be ESM.
```suggestion
export default {
```

File: packages/website/tsconfig.json [hunk 9]
```diff
@@ -1,17 +1,7 @@
...
     "outDir": "./dist",
-    "tsBuildInfoFile": "./dist/.tsbuildinfo",
-    "lib": ["ESNext", "DOM", "DOM.Iterable"]
-  },
-  "include": ["src"],
-  "references": [
-    {
-      "path": "../waku"
-    }
-  ]
```
Comment (dai-shi) [inline packages/website/tsconfig.json:16]: I'm not 100% sure, but we should probably keep "include" and "references".

File: packages/website/tsconfig.json [hunk 10]
```diff
@@ -1,12 +1,8 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "declaration": true,
-    "emitDeclarationOnly": true,
-    "rootDir": "./src",
     "outDir": "./dist",
-    "tsBuildInfoFile": "./dist/.tsbuildinfo",
-    "lib": ["ESNext", "DOM", "DOM.Iterable"]
+    "rootDir": "./src"
```
Comment (dai-shi) [inline packages/website/tsconfig.json:5]: Please move this before "outDir" to make the diff minimal.

File: packages/website/src/components/credits.tsx [hunk 11]
```diff
@@ -0,0 +1,56 @@
+export const Credits = () => {
+  if (process.env.VITE_SHOW_CREDITS !== "1") return null;
```
Comment (dai-shi) [inline packages/website/src/components/credits.tsx:2]:
```suggestion
  if (process.env.VITE_SHOW_CREDITS !== "YES") return null;
```

File: packages/website/tsconfig.json [hunk 12]
```diff
@@ -1,12 +1,8 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "declaration": true,
-    "emitDeclarationOnly": true,
     "rootDir": "./src",
-    "outDir": "./dist",
-    "tsBuildInfoFile": "./dist/.tsbuildinfo",
-    "lib": ["ESNext", "DOM", "DOM.Iterable"]
+    "outDir": "./dist"
```
Comment (dai-shi) [inline packages/website/tsconfig.json:5]: It feels good to me, but let me mention @himself65 to check if this can cause any problems.

**Review by dai-shi — APPROVED** [review]
LGTM

File: packages/website/tsconfig.json (hunk 12 above)
Comment (himself65) [reply packages/website/tsconfig.json:5]: I think we should keep `tsBuildInfoFile` otherwise it will generate to the package dir

Comment (himself65) [reply packages/website/tsconfig.json:5]: Like this 
image

Comment (dai-shi) [reply packages/website/tsconfig.json:5]: I see. Is it for incremental build? I'd put it outside `dist`. Let me try.

Comment (himself65) [reply packages/website/tsconfig.json:5]: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html

We could put it anywhere. It's just a cache file for incremental build.

Comment (dai-shi) [reply packages/website/tsconfig.json:5]: Yeah, I don't think it's necessary in dist: https://unpkg.com/browse/waku@0.16.0/dist/.tsbuildinfo
